local_server/data_common/data_adaptor.py

```diff
@@ -283,17 +284,16 @@ def check_new_gene_sets(self, args, context=None):
            will generate a warning and the symbol removed.
         3. Duplicate gene symbols are silently de-duped.
         """
-        (genesets, tid) = args
         messagefn = context["messagefn"] if context else (lambda x: None)
 
         # accept genesets args as either the internal (dict) or REST (list) format,
         # as they are identical except for the dict being keyed by geneset_name.
         if type(genesets) not in (dict, list):
             raise ValueError("Genesets must be either dict or list.")
-        genesets = genesets if type(genesets) == list else genesets.values()
+        genesets_iterable = genesets if type(genesets) == list else genesets.values()
```

Comment:
This is very much a nit but I believe the more Pythonic way to check for type is `isinstance(genesets, list)`. Noticed it in a few places, but up to you whether to switch or not. Just a nit.

Reply:
will fix.  I assume I should do this for all of the other type(x) is Y tests? (there are many :-)

Reply:
I'm happy just not _adding_ additional `type(x) == type` lines in this PR :) Don't have to switch everything right now.

Reply:
oops - I just did a bunch. Will commit them and stop!